code/modules/power/apc.dm

```diff
@@ -581,6 +581,32 @@
 			return TRUE
 
 /obj/machinery/power/apc/attackby(obj/item/W, mob/living/user, params)
+	if(istype(W, /obj/item/kitchen))
+		var/metal = FALSE
+		var/shock_source = FALSE
+
+		if(LAZYLEN(W.custom_materials)) //This prevents wooden rolling pins from shocking the user
+			if(W.custom_materials[GET_MATERIAL_REF(/datum/material/iron)])
+				metal += W.custom_materials[GET_MATERIAL_REF(/datum/material/iron)]
+
+		if(cell || terminal) //The mob gets shocked by whichever powersource has the most electricity
+			if(cell && terminal)
+				if(cell.charge > terminal.powernet.avail)
+					shock_source = cell
+				else
+					shock_source = terminal.powernet
+			if(!terminal)
+				shock_source = cell
+			if(!cell)
+				shock_source = terminal.powernet
+
+		if(shock_source && metal && !user.combat_mode && (panel_open || opened)) //Now you're cooking with electricity
+			if(electrocute_mob(user, shock_source, src, siemens_coeff = 1, dist_check = TRUE))//People with insulated gloves just attack the APC normally. They're just short of magical anyway
```

Comment:
I'd suggest giving them a special message instead. Maybe something about proving all of those naysayers wrong?

Reply:
Why do you need a message instead of frying and not hitting the APC and not frying and hitting the apc?

Reply:
because you're clicking on it while NOT in combat mode, so you're not trying to bash it, you're trying to stick a kitchen implement into it.

Reply:
You hit it with the implement even if you're not on combat mode unless it shocks you, which stops you from doing so.
This is consistent with all of our other "non-harmies but hit anyway" controls